common/lib/driver-definitions/api-report/driver-definitions.api.md

```diff
@@ -149,16 +150,16 @@ export interface IDocumentServicePolicies {
 
 // @public
 export interface IDocumentStorageService extends Partial<IDisposable> {
-    createBlob(file: ArrayBufferLike): Promise<ICreateBlobResponse>;
-    downloadSummary(handle: ISummaryHandle): Promise<ISummaryTree>;
-    getSnapshotTree(version?: IVersion): Promise<ISnapshotTree | null>;
-    getVersions(versionId: string | null, count: number): Promise<IVersion[]>;
+    createBlob(file: ArrayBufferLike, fetchReason?: string): Promise<ICreateBlobResponse>;
```

Comment:
Some thoughts:
1. I do not think reason makes sense for many of these. for example createBlob() - I do not think we can ask users of FF to provide it (what if it has customer PII?) and FF can't provide meaningful data here as well. readBlob() is sort of similar (though I can see us using reason = "user" | "gcBlob" | "potentiallyDeletedBlob" - something like that, i.e. not end-user controlled, but that still violates the point I'll make below).
   - Even in places where FF could provide reason (getVersions), the API would need to clearly document how it's used, such that both parties (driver & user of driver) clearly understand impact (i.e. some of these are exposed to runtime, are we sure components will not start pushing PII data?)
   - In general, I think it's Ok to have reason on API only if it's not exposed to runtime, i.e. it's API used solely at loader level.
2. Some APIs I'd rather see deleted that keep adding stuff to it (downloadSummary / write - it's not used as far as I know).
3. On some - it should does not make sense (uploadSummaryWithContext - if we need to communicate something, it's better be part of ISummaryContext, and on top of it there is only one reason to call it :))
4. Maybe this point is moot after all of the above: "fetch reason" terminology does not make sense for "push" APIs (like summary upload). The API semantic is to push, not fetch, so word "fetch" is misleading (and refers to implementation detail).

I think obvious thing to ask - do we have to provide that info for all calls, or 99% of benefits is covered by a few that do not complicate API?

That all said, I'd love to see "fetch reason" propagated for ops fetches - it's indeed a pain even for us, looking at aggregated driver telemetry and not being able to separate reasons.

Reply:
Makes sense. I wasn't sure about the PII part but I added in all as for future usability if need be. But it seems that is not fine and not required. So removing these and just leaving for ops fetch. Also removing for getVersions() and getSnapshotTree() as those 2 apis are used in runtime as well.